quantum/quantum_task_state_handler.h

```diff
@@ -0,0 +1,104 @@
+/*
+** Copyright 2022 Bloomberg Finance L.P.
+**
+** Licensed under the Apache License, Version 2.0 (the "License");
+** you may not use this file except in compliance with the License.
+** You may obtain a copy of the License at
+**
+**     http://www.apache.org/licenses/LICENSE-2.0
+**
+** Unless required by applicable law or agreed to in writing, software
+** distributed under the License is distributed on an "AS IS" BASIS,
+** WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+** See the License for the specific language governing permissions and
+** limitations under the License.
+*/
+#ifndef BLOOMBERG_QUANTUM_TASK_STATE_HANDLER_H
+#define BLOOMBERG_QUANTUM_TASK_STATE_HANDLER_H
+
+#include <functional>
+
+namespace Bloomberg {
+namespace quantum {
+
+//==============================================================================================
+//                                       enum TaskType
+//==============================================================================================
+/// @enum TaskType
+/// @brief Defines the types of the tasks.
+/// @note This enum should be used to specify the type of the task
+///       for which a task state handler is called
+enum class TaskType : int {
+    None        = 0,      ///< No task type
+    Coroutine   = 1 << 0, ///< Coroutine
+    IoTask      = 1 << 1  ///< Blocking IO (or long running) task
```

Comment:
All = Coroutine  | IoTask
?

Reply:
It can be done later if needed. I don't know the case that handler will be called for both `IoTask` and `Coroutine`.